### PR TITLE
Make github username in function name to be lowercase

### PR DESCRIPTION
**Pull request by martindekov**
Make github username in function names to be lower so
the of-router can access it

Signed-off-by: Martin Dekov (VMware) <mdekov@vmware.com>

## Description

Changed github owner to be build with lowercase username i.e. name of function created with faas-cli `gofunc` name of github owner `MartinDekovWat` is now `martindekovwat-gofunc` unlike before `MartinDekovWat-gofunc`

Part of #119 

## How Has This Been Tested?

Tested three cases:

Lowercase github username: `martindekov` pushed function `goforfun` to the cloud
```
$ curl "http://martindekov.md.team-serverless.xyz:8082/gocloudfunc"
Wssp wat
```

Mixed case github username: `MartinDekovWat` pushed function `goforfun` in repo `exmpl`
the relevent field of github-status logs:
```
github-status.1.i08lngosblho@FaasSwarm    | 2018/09/03 15:20:23 Status: success, Context: goforfun, Service: MartinDekovWat-exmpl, GitHub AppID: 312235, Repo: exmpl, Owner: MartinDekovWat
```
We can curl with of-router and it works 
```
$ curl "http://martindekovwat.md.team-serverless.xyz:8082/goforfun"
 yow was da way to GROW?
```
There is the status of the commits `https://github.com/MartinDekovWat/exmpl/commits/master` it works

MIxed case github username: `MartinDekovWat` pushed function `thesensitivefunc` in repo `IAmSensitiveFunction`
```
$ curl "http://martindekovwat.md.team-serverless.xyz:8082/thesensitivefunc"
Want to see logs working
```
We can see `github-push` did its job in commit statuses `https://github.com/MartinDekovWat/IAmSensitiveFunction/commits/master`

renamed function in stack.yml from `thesensitivefunc` to `theSensitiveFunc`
in here `https://github.com/MartinDekovWat/IAmSensitiveFunction/blob/master/stack.yml`
the cloud function name changed to `martindekovwat-theSensitiveFunc` from `martindekovwat-thesensitivefunc`
## Checklist:

I have:

- [ ] updated the documentation if required
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: buildshiprun/handler.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func Handle(req []byte) string {
 		Source: "buildshiprun",
 	}
 
-	serviceValue := fmt.Sprintf("%s-%s", event.Owner, event.Service)
+	serviceValue := fmt.Sprintf("%s-%s", strings.ToLower(event.Owner), event.Service)
```
Comment (alexellis) [inline buildshiprun/handler.go]: Let's create a method for this so that we can track it and test it.

File: github-push/handler.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func Handle(req []byte) string {
 		return msg
 	}
 
-	serviceValue := fmt.Sprintf("%s-%s", pushEvent.Repository.Owner.Login, pushEvent.Repository.Name)
+	serviceValue := fmt.Sprintf("%s-%s", strings.ToLower(pushEvent.Repository.Owner.Login), pushEvent.Repository.Name)
```
Comment (alexellis) [inline github-push/handler.go]: Looks like the 2nd occurrence. So let's break this out into the sdk package.

File: github-status/handler.go [hunk 3]
```diff
@@ -113,7 +113,7 @@ func buildPublicStatusURL(status string, statusContext string, event *sdk.Event)
 				}
 				// for success status if gateway's public url id set the deployed
 				// function url is used in the commit status
-				serviceValue := fmt.Sprintf("%s-%s", event.Owner, event.Service)
+				serviceValue := fmt.Sprintf("%s-%s", strings.ToLower(event.Owner), event.Service)
```
Comment (alexellis) [inline github-status/handler.go]: 3rd occurrence. Let's break it out?

Comment (martindekov) [reply github-status/handler.go]: Good idea

File: sdk/events.go [hunk 4]
```diff
@@ -56,3 +61,7 @@ func BuildEventFromPushEvent(pushEvent PushEvent) *Event {
 
 	return &info
 }
+
+func ServiceName(userName, functionName string) string {
```
Comment (alexellis) [inline sdk/events.go]: Let's have it as `FormatServiceName` please?

userName could be `owner`

File: github-push/vendor/github.com/openfaas/openfaas-cloud/sdk/events.go [hunk 5]
```diff
@@ -1,5 +1,10 @@
 package sdk
 
+import (
```
Comment (alexellis) [inline github-push/vendor/github.com/openfaas/openfaas-cloud/sdk/events.go]: events.go is probably the wrong file.

How about something else .go?

Comment (martindekov) [reply github-push/vendor/github.com/openfaas/openfaas-cloud/sdk/events.go]: Yeah should I create separate file for that like service.go?

**Comment by alexellis** [comment]
Brilliant 👍

**Pull request merged**